### PR TITLE
#163518683 User get verification link

### PR DESCRIPTION
**Pull request by niomwungeri-fabrice**
#### What does this PR do?
Email verification 
#### Description of Task to be completed?
After a user has been created successfully the platform sends the email via email,
the user is able to click on the link in order to get access to other services like creating an article.
The is an option of resending the email if the email wasn's sent on time.
EndPoints:
* `POST api/v1/users` for creating an account to the system, and receive a verification email.
* `GET api/v1/users/confirm/:auth_token` for email verification.
* `PUT api/v1/users/resend` for resending email verification.
#### How should this be manually tested?
* Fetch the branch `ft-user-get-verification-link-163518683`
* Checkout the branch `ft-user-get-verification-link-163518683`
* Setup your database. 
* Run tests `yarn run test`
* Start the server `yarn run server`
* Create a user `POST: /api/v1/users` with `email, password, and username`
* Go to your email click on the `verify my email` which was sent while registering it will trigger this endPoint `GET api/v1/users/confirm/:auth_token` using Postman client. 
* You should see a User verified successfully. 
* If the email didn't come you could use this endpoint to resend the email `PUT api/v1/users/resend` you only provid email as follow:
`
{
	"email":"example@gmail.com"
}
`
#### What are the relevant pivotal tracker stories?
163518683

### CODE REVIEW

**Review by YvesIraguha — APPROVED** [review]
Correct repetition errors in the description of this PR under the section `Description of Task to be completed?`

File: migrations/20190211132848-add-isVerfied-on-users.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    /*
+<<<<<<< HEAD
```
Comment (abayo-luc) [inline migrations/20190211132848-add-isVerfied-on-users.js:6]: Remove conflict in this file

File: middlewares/validators/signup.validator.js~HEAD [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import { isEmpty, isEmailValid, isAlphanumeric } from "../../helpers/funcValidators";
+
```
Comment (abayo-luc) [inline middlewares/validators/signup.validator.js~HEAD:2]: This file seems to be a copy of something else, please double check.

File: migrations/seeders/20190130170205-create-user.js [hunk 3]
```diff
@@ -9,7 +9,28 @@ module.exports = {
           id: "dfe7fbc2-289d-11e9-b210-d663bd873d97",
           email: "test@ah.com",
           password: "password",
+<<<<<<< HEAD
```
Comment (abayo-luc) [inline migrations/seeders/20190130170205-create-user.js]: Resolve conflict @niomwungeri-fabrice

File: tests/users/emailConfirmation.test.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import chai, { expect, should } from "chai";
+import app from "../../index";
+import models from "../../models";
+import { data } from "../../helpers/data";
+
+const { dummyUser3 } = data;
+chai.use(chaiHttp);
+should();
+let userToken;
+const invalidToken =
```
Comment (abayo-luc) [inline tests/users/emailConfirmation.test.js]: You can save this kind of data in ```.../../helpers/data```

**Review by abayo-luc — CHANGES_REQUESTED** [review]

**Review by abayo-luc — APPROVED** [review]

**Pull request closed without merging**